test/e2e/perftype/perftype.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package perftype
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// The following performance data structures are generalized and well-formatted.
+// They can be pretty printed in json format and be analyzed by other performance
+// analyzing tools, such as Perfdash (k8s.io/contrib/perfdash).
+
+// Histogram is a map from bucket to real data point (e.g. "Perc90" -> 23.5). Notice
+// that all data items with the same tag conbination should have the same buckets.
+type Histogram map[string]float64
+
+// DataItem is the data point.
+type DataItem struct {
+	// Data is the real data.
+	Data Histogram `json:"data"`
+	// Unit is the data unit. Notice that all data items with the same tag combination
+	// should have the same unit.
+	Unit string `json:"unit"`
```

Comment:
Maybe embed it into Data structure to make it clear? I.e. make type <NewHistogramName> struct { data map[string]float64, unit string }